packages/contracts/src/contracts/current/protocol/Exchange/MixinMatchOrders.sol

```diff
@@ -0,0 +1,285 @@
+/*
+  Copyright 2018 ZeroEx Intl.
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+pragma solidity ^0.4.21;
+pragma experimental ABIEncoderV2;
+
+import "./mixins/MExchangeCore.sol";
+import "./mixins/MMatchOrders.sol";
+import "./mixins/MSettlement.sol";
+import "./mixins/MTransactions.sol";
+import "../../utils/SafeMath/SafeMath.sol";
+import "./LibOrder.sol";
+import "./LibStatus.sol";
+import "./LibPartialAmount.sol";
+import "../../utils/LibBytes/LibBytes.sol";
+
+contract MixinMatchOrders is
+    SafeMath,
+    LibBytes,
+    LibStatus,
+    LibOrder,
+    LibPartialAmount,
+    MExchangeCore,
+    MMatchOrders,
+    MSettlement,
+    MTransactions
+    {
+
+    /// @dev Validates context for matchOrders. Succeeds or throws.
+    /// @param leftOrder First order to match.
+    /// @param rightOrder Second order to match.
+    function validateMatchOrdersContextOrRevert(
+        Order memory leftOrder,
+        Order memory rightOrder)
+        internal
+    {
+        // The leftOrder maker asset must be the same as the rightOrder taker asset.
+        require(areBytesEqual(leftOrder.makerAssetData, rightOrder.takerAssetData));
```

Comment:
Rather than checking that these are equal, it would be more efficient to _assume_ that they are equal (less calldata to pass in). 

This can be done by passing in null byte arrays for `makerAssetData` and `takerAssetData` in `rightOrder` OR having `rightOrder` be a different struct that excludes these fields (slightly more efficient).

Reply:
That assumption sounds like it could be dangerous, no? I'm also not positive it would make this call that much more efficient. This is an `internal` method that receives memory addresses as function parameters, opposed to entire structs.